### PR TITLE
Backporting PR for allocating in the old space

### PR DESCRIPTION
**Pull request by jordanmontt**
Backports https://github.com/pharo-project/pharo-vm/pull/701/files into `pharo-10` branch

### CODE REVIEW

File: smalltalksrc/VMMaker/SpurMemoryManager.class.st [hunk 1]
```diff
@@ -6935,24 +6945,42 @@ SpurMemoryManager >> instantiateClass: classObj indexableSize: nElements [
...
 SpurMemoryManager >> instantiateClass: classObj isPinned: isPinned [
+
+	^ self instantiateClass: classObj isPinned: isPinned isOldSpace: false
+]
+
+{ #category : #instantiation }
+SpurMemoryManager >> instantiateClass: classObj isPinned: isPinned isOldSpace: isOldSpace [
 	| instSpec classFormat numSlots classIndex newObj |
-	classFormat := self formatOfClass: classObj.
+	classFormat := self formatOfClassSafe: classObj.
```
Comment (jordanmontt) [inline smalltalksrc/VMMaker/SpurMemoryManager.class.st]: There is this change that I didn't do… I suppose that is a change that was done on `pharo-12`?

**Review by guillep — APPROVED** [review]
Ok to me!

**Comment by guillep** [comment]
It does not compile:

```
/Users/lse/Workspace/workspace/pharo-vm_backport-pr-new-old/build/generated/64/vm/src/gcc3x-cointerp.c:47568:16: error: implicit declaration of function 'formatOfClassSafe' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

        classFormat = formatOfClassSafe(classObj);
```

We need to use formatOfClass, formatOfClassSafe does not exist in this branch.

File: smalltalksrc/VMMaker/SpurMemoryManager.class.st (hunk 1 above)
Comment (guillep) [reply smalltalksrc/VMMaker/SpurMemoryManager.class.st]: Yes, we need to *not* do this change.

File: smalltalksrc/VMMaker/SpurMemoryManager.class.st [hunk 2]
```diff
@@ -6935,22 +6945,35 @@ SpurMemoryManager >> instantiateClass: classObj indexableSize: nElements [
...
+]
+
+{ #category : #instantiation }
+SpurMemoryManager >> instantiateClass: classObj isPinned: isPinned isOldSpace: isOldSpace [
 	| instSpec classFormat numSlots classIndex newObj |
 	classFormat := self formatOfClass: classObj.
 	instSpec := self instSpecOfClassFormat: classFormat.
 	(self isFixedSizePointerFormat: instSpec) ifFalse:
-		[^nil].
+		 ^nil].
```
Comment (guillep) [inline smalltalksrc/VMMaker/SpurMemoryManager.class.st]: This is not pharo syntax, it's making the CI fail :)

Comment (jordanmontt) [reply smalltalksrc/VMMaker/SpurMemoryManager.class.st]: hehe i should not edit Pharo code using VSC

Comment (guillep) [reply smalltalksrc/VMMaker/SpurMemoryManager.class.st]: ;D

**Comment by jordanmontt** [comment]
It should be good by now!

**Pull request merged**